src/IECore/VectorTypedData.cpp

```diff
@@ -220,6 +220,7 @@ IE_CORE_DEFINEIMATHVECTORTYPEDDATASPECIALISATION( Color4fVectorData, Color4fVect
 
 IECORE_RUNTIMETYPED_DEFINETEMPLATESPECIALISATION( StringVectorData, StringVectorDataTypeId )
 IE_CORE_DEFINENOBASEVECTORTYPEDDATAIOSPECIALISATION( StringVectorData )
+IE_CORE_DEFINEVECTORTYPEDDATATRAITSSPECIALIZATION ( StringVectorData )
```

Comment:
I'm not sure about this... By defining `IE_CORE_DEFINEVECTORTYPEDDATATRAITSSPECIALIZATION` we're saying `StringVectorData` has a base type that can be accessed directly, but on the line above we're still choosing the "NOBASE" variant of the IO specialisation.

I have to confess to not knowing much about this stuff... maybe we're better to wait for @johnhaddon to get a third eye on it.

Reply:
Looks good to me.

I think Andrew's right about the use of the NOBASE variant being odd, but it was odd before, because StringVectorData has always [defined a base type](https://github.com/ImageEngine/cortex/blob/master/include/IECore/VectorTypedData.h#L67) publicly. It actually looks like IE_CORE_DEFINENOBASEVECTORTYPEDDATAIOSPECIALISATION is only used for StringVectorData, and the IE_CORE_DEFINEBASEVECTORTYPEDDATAIOSPECIALISATION version would work just fine in its place. So perhaps we should simplify things and remove IE_CORE_DEFINENOBASEVECTORTYPEDDATAIOSPECIALISATION entirely?

Reply:
Removed the macro in 19ad696. @johnhaddon, I don't really understand the reasons for why StringVectorData was different but also it's probably not worth going into now it's deleted.

Reply:
As far as I can see, there wasn't any good reason for it to be different in the first place...